jgrapht-core/src/test/java/org/jgrapht/alg/spanning/MinimumSpanningTreeTest.java

```diff
@@ -17,96 +17,71 @@
+    public static final String G = "G";
+    public static final String H = "H";
 
     // ~ Instance fields --------------------------------------------------------
 
-    private DefaultWeightedEdge AB;
-    private DefaultWeightedEdge AC;
-    private DefaultWeightedEdge BD;
-    private DefaultWeightedEdge DE;
-    private DefaultWeightedEdge EG;
-    private DefaultWeightedEdge GH;
-    private DefaultWeightedEdge FH;
+    public static DefaultWeightedEdge AB;
+    public static DefaultWeightedEdge AC;
+    public static DefaultWeightedEdge BD;
+    public static DefaultWeightedEdge DE;
+    public static DefaultWeightedEdge EG;
+    public static DefaultWeightedEdge GH;
+    public static DefaultWeightedEdge FH;
 
     // ~ Methods ----------------------------------------------------------------
 
     @Test
-    public void testKruskal()
-    {
-
-        testMinimumSpanningTreeBuilding(
-            new KruskalMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleConnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, DE), 15.0);
-
-        testMinimumSpanningTreeBuilding(
-            new KruskalMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
+    public void testSimpleDisconnectedWeightedGraph(){
+        testMinimumSpanningTreeBuilding(createSolver(createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
+                Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
     }
 
     @Test
-    public void testPrim()
-    {
-        testMinimumSpanningTreeBuilding(
-            new PrimMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleConnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, DE), 15.0);
-
-        testMinimumSpanningTreeBuilding(
-            new PrimMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
-    }
-
-    @Test
-    public void testBoruvska()
-    {
-        testMinimumSpanningTreeBuilding(
-            new BoruvkaMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleConnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, DE), 15.0);
-
-        testMinimumSpanningTreeBuilding(
-            new BoruvkaMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
+    public void testSimpleConnectedWeightedGraph() {
+        testMinimumSpanningTreeBuilding(createSolver(createSimpleConnectedWeightedGraph()).getSpanningTree(),
+                Arrays.asList(AB, AC, BD, DE), 15.0);
     }
 
     @Test
     public void testRandomInstances()
```

Comment:
This random test needs to be moved to the specific implementing classes. Now this exact same test is executed 4 times (PrimTest, PrimDenseTest, ...). Since the test is seeded with the same seed, it wouldn't make sense to execute it 4 times :)

Reply:
Shouldn't that test be executed once for each implementation?